tools/cli/commands/generate.js

```diff
@@ -232,26 +236,107 @@ export function getQuestions( type ) {
+
+	createSkeleton( pluralize( answers.type ), pkgDir, answers.name );
+
+	// Generate the package.json file
+	const packageJson = readPackageJson( project );
+	createPackageJson( packageJson, answers );
+	writePackageJson( project, packageJson, pkgDir );
+
+	// Generate the composer.json file
+	const composerJson = readComposerJson( project );
+	createComposerJson( composerJson, answers );
+	writeComposerJson( project, composerJson, pkgDir );
+
+	return packageJson;
+}
+
+/**
+ * Generate a plugin based on questions passed to it.
+ *
+ * @todo REMOVE EXPORT. ONLY FOR TESTING.
+ *
+ * @param {object} answers - Answers from questions.
+ *
+ * @returns {object} package.json object. TEMPORARY FOR TESTING.
+ */
+export function generatePlugin( answers = { name: 'test', description: 'n/a', buildScripts: [] } ) {
+	const project = 'plugins/' + answers.name;
+	const pluginDir = path.join( __dirname, '../../..', 'projects/plugins', answers.name );
+
+	// Copy the skeletons over
+	createSkeleton( pluralize( answers.type ), pluginDir, answers.name );
+
+	// Generate the package.json file
+	const packageJson = readPackageJson( project );
+	createPackageJson( packageJson, answers );
+	writePackageJson( project, packageJson, pluginDir );
+
+	// Generate the composer.json file
+	const composerJson = readComposerJson( project );
+	createComposerJson( composerJson, answers );
+	writeComposerJson( project, composerJson, pluginDir );
+
+	// Write header to plugin's main file.
+	const headerContent = createPluginHeader( answers );
+	writeToFile( pluginDir + `/${ answers.name }.php`, headerContent );
+
+	// Fill in the README.md file
+	writeToFile( pluginDir + '/README.md', `# ${ answers.name }` );
+
+	return packageJson;
+}
+
+/**
+ * Create skeleton files for project
+ *
+ * @todo REMOVE EXPORT. ONLY FOR TESTING.
+ *
+ * @param {string} type - Type of project.
+ * @param {string} dir - Directory of new project.
+ * @param {string} name - Name of new project.
+ *
+ */
+export function createSkeleton( type, dir, name ) {
 	const skeletonDir = path.join( __dirname, '../skeletons' );
 
 	// Copy the skeletons over.
 	try {
-		mergeDirs( path.join( skeletonDir, '/common' ), pkgDir );
-		mergeDirs( path.join( skeletonDir, '/packages' ), pkgDir );
+		mergeDirs( path.join( skeletonDir, '/common' ), dir, name );
+		mergeDirs( path.join( skeletonDir, '/' + type ), dir, name );
 	} catch ( e ) {
 		console.error( e );
 	}
-	const project = 'packages/' + answers.name;
+	return;
+}
 
-	// Generate the package.json file
-	const packageJson = readPackageJson( project );
+/**
+ * Create package.json for project
+ *
+ * @todo REMOVE EXPORT. ONLY FOR TESTING.
+ *
+ * @param {object} packageJson - The parsed skeleton JSON package file for the project.
+ * @param {object} answers - Answers returned for project creation.
+ *
+ */
+export function createPackageJson( packageJson, answers ) {
 	packageJson.description = answers.description;
+	return;
```

Comment:
what is this actually doing with the return here?

Reply:
> what is this actually doing with the return here?

It's returning because self taught Treehouse acolyte Steve hasn't rid himself of the habit of dropping return at the end of functions that don't return anything.